test/kb_Metrics_server_test.py

```diff
@@ -232,248 +473,447 @@ def test_db_controller_contrusctor(self):
-        print(pformat(ret[0]['metrics_result']))
+        ret = self.getImpl().get_user_details(self.getContext(), m_params)
+        users = ret[0]['metrics_result']
+        self.assertEqual(len(users), 33)
 
     # Uncomment to skip this test
-    @unittest.skip("skipped test_run_get_user_counts_per_day")
-    def test_run_get_user_counts_per_day(self):
+    # @unittest.skip("skipped test_run_MetricsImpl_get_user_counts_per_day")
+    def test_run_MetricsImpl_get_user_counts_per_day(self):
         m_params = {
-            'user_ids': [],  # ['qzhang'],#'user_ids': [],
-            #'epoch_range':(1420083768000, 1505876263000)#(datetime.datetime(2015, 1, 1), datetime.datetime(2017,9,20))
-            'epoch_range': (datetime.datetime(2017, 1, 1), datetime.datetime(2017, 6, 28))
+            'epoch_range': (datetime.datetime(2018, 1, 1), datetime.datetime(2018, 3, 31))
         }
-        # Second, call your implementation
         ret = self.getImpl().get_user_counts_per_day(self.getContext(), m_params)
-        print("Total number of records get_user_counts_per_day returned=" +
-              str(len(ret[0]['metrics_result'])))
-        print(pformat(ret[0]['metrics_result']))
+        # testing (excluding kbstaff by default)
+        users = ret[0]['metrics_result']
+        self.assertEqual(len(users), 56)
+        self.assertIn('numOfUsers', users[0])
+        self.assertIn('yyyy-mm-dd', users[0])
+        self.assertEqual(users[0]['yyyy-mm-dd'], '2018-1-1')
+        self.assertEqual(users[0]['numOfUsers'], 1)
+        self.assertEqual(users[1]['numOfUsers'], 1)
+        self.assertEqual(users[2]['numOfUsers'], 2)
+        self.assertEqual(users[3]['numOfUsers'], 2)
 
     # Uncomment to skip this test
-    @unittest.skip("skipped test_run_get_user_narratives")
-    def test_run_get_user_narratives(self):
+    # @unittest.skip("skipped test_run_MetricsImpl_update_metrics")
+    def test_run_MetricsImpl_update_metrics(self):
         m_params = {
-            'user_ids': ['psdehal'],  # ['qzhang'],#'user_ids': [],
-            #'epoch_range':(1420083768000, 1505876263000)#(datetime.datetime(2015, 1, 1), datetime.datetime(2017,9,20))
-            'epoch_range': (datetime.datetime(2014, 1, 1), datetime.datetime(2018, 2, 28))
+            'user_ids': [],
+            'epoch_range': (datetime.datetime(2018, 2, 28), datetime.datetime(2018, 3, 31))
         }
-        # Second, call your implementation
-        ret = self.getImpl().get_user_narratives(self.getContext(), m_params)
-        #print("Total number of records get_user_narratives returned="+str(len(ret[0]['metrics_result'])))
-        print(pformat(ret[0]['metrics_result']))
+        ret = self.getImpl().update_metrics(self.getContext(), m_params)
+        upds = ret[0]['metrics_result']
+        self.assertIn('user_updates', upds)
+        self.assertIn('activity_updates', upds)
+        self.assertIn('narrative_updates', upds)
+        self.assertTrue(upds['user_updates'] >= 0)
```

Comment:
should upds['user_updates'] == 0 be considered as failure?

Reply:
more tests added

Reply:
I fixed ALL pep8 errors.